### PR TITLE
Makefile: add bin/*, resolve and watch-resolve target 🚕

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

Those targets are there to build binaries using `go` and the images
using `ko resolve` without applying the changes.

This also adds a `generated` target.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/kind misc
/cc @afrittoli @sbwsg 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Add bin/*, resolve, watch-resolve and generated target to the Makefile
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,13 +35,31 @@ $(BIN)/%: | $(BIN) ; $(info $(M) building $(PACKAGE)…)
...
+bin/%: cmd/% FORCE
+	$(GO) build -mod=vendor $(LDFLAGS) -v -o $@ ./$<
+
 KO = $(BIN)/ko
 $(BIN)/ko: PACKAGE=github.com/google/ko/cmd/ko
 
 .PHONY: apply
 apply: | $(KO) ; $(info $(M) ko apply -f config/) @ ## Apply config to the current cluster
-	$Q $(KO) apply --strict -f config
+	$Q $(KO) apply -f config
```
Comment (chmouel) [inline Makefile:48]: why did you remove the --strict, is that intentional?

Comment (vdemeester) [reply Makefile:48]: Yes and no :sweat_smile: With `--strict`, it doesn't build any image for me (for some weird reason).

Comment (mattmoor) [reply Makefile:48]: `--strict` should only filter out things prefixed with `ko://`, so if you are seeing stuff that is prefixed not building we should talk :)

Comment (vdemeester) [reply Makefile:48]: @mattmoor just realized why… `ko://` got temporarly remove because of https://github.com/google/ko/issues/164 I think :stuck_out_tongue_closed_eyes: hence why it doesn't build a thing :stuck_out_tongue_closed_eyes:

Comment (afrittoli) [reply Makefile:48]: maybe you could add the `strict` as an option?
